collections-plan.md

```diff
@@ -0,0 +1,102 @@
+# Transactions plan
+A list of requirements to support transactions from one collection to another.
+
+### Requirements
+* Change defaults code to allow any defaults and not to force quantity
+* Update local storage to support items being added and removed from collections
+* Implement `moveItem` method 
+
+### Move command description
+
+`moveItem(object: transaction)`
+
+```
+transaction:
+{
+    from: String, //required
+    to: String, //required
+    key:  String, //required
+    decrement: Integer, //default 1
+    fromState: String, //optional
+    toState: String, //optional
+}
+```
+
+## Example Scenarios
+
+#### Scenario 1
+* Shop has 1 shield
+* Player buys shield
+* Shield is added to inventory
+* Shield shows in shop as “owned”
+
+**start collection states:**
+shop [{key: "shield"}]
+inventory []
+
+**Command**
+`collections.moveItem({key: "shield", from: "shop", to: "player", fromState: "owned"])`
+
+**end collection state:**
+shop [{key: "shield", state: "owned"}]
+inventory [{key: "shield"}]
+
+#### Scenario 2
+* Shop has 5 bananas
+* Player buys 1 banana
+* 1 banana is added to inventory
+* 4 bananas show in shop
+
+**start collection states:**
+shop [{key: "banana", qty: 5}]
+inventory []
+
+**Command**
+`collections.moveItem({key: "banana", from: "shop", to: "player"})`
```

Comment:
guessing this is a `to: "inventory"` too?

Reply:
lol. Yep.!